superset/hive_query.py

```diff
@@ -95,49 +96,114 @@ def get_partitioned_whereclause(_st, _en, gran_seconds, time_partitions):
 
     return where_clause
 
+# Custom query for Hive Partitons (Date Based)
+def where_clause_date_based(_st, _en, date_partitions, grain):
+    time_seq = list()
+
+    if grain == 0:
+        # Partition = Date
+        part_time_seq = list()
```

Comment:
This seems redundant in this if block

Reply:
Okay, I'll shift it.

Reply:
Done that. Also committed some changes which handle some corner cases (important).